package/carbonio-proxy

```diff
@@ -1,83 +1,68 @@
-echo -n "insert the cluster credential password: "
+# to avoid re-asking for the password
+echo -n "Insert the cluster credential password: "
+# shellcheck disable=SC2155
 export CONSUL_HTTP_TOKEN=$(service-discover bootstrap-token --setup)
 EXIT_CODE="$?"
 echo ""
 if [[ "${EXIT_CODE}" != "0" ]]; then
-  echo "cannot access to bootstrap token"
+  echo "Cannot access to bootstrap token"
   exit 1;
 fi
-# limit secret visibility as much as possible
+# Limit secret visibility as much as possible
 export -n SETUP_CONSUL_TOKEN
 
-POLICY_NAME='carbonio-proxy-policy'
-POLICY_DESCRIPTION='Carbonio Proxy service policy for config generation and for sidecar proxy'
-POLICY_RULES="$(cat <<EOF
-"key_prefix" = {
-  "carbonio-proxy/" = {
-    "policy" = "read"
-  }
-}
-"node_prefix" = {
-  "" = {
-    "policy" = "read"
-  }
-}
-"service" = {
-  "carbonio-proxy" = {
-    "policy" = "write"
-  }
-  "carbonio-proxy-sidecar-proxy" = {
-    "policy" = "write"
-  }
-}
-EOF
-)"
+POLICY_NAME="${SERVICE_NAME}-policy"
+POLICY_DESCRIPTION="Carbonio ${SERVICE} service policy for service and sidecar proxy"
 
-# create or update policy for the specific service (this will be shared across cluster)
-consul acl policy create -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules "${POLICY_RULES}" >/dev/null 2>&1
+# Create or update policy for the specific service (this will be shared across cluster)
+consul acl policy create -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules @"${SERVICE_BASE_DIR}/policies-${SERVICE}.json" >/dev/null 2>&1
+# shellcheck disable=SC2181
 if [[ "$?" != "0" ]]; then
-    consul acl policy update -no-merge -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules "${POLICY_RULES}"
+    consul acl policy update -no-merge -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules @"${SERVICE_BASE_DIR}/policies-${SERVICE}.json"
     if [[ "$?" != "0" ]]; then
       echo "Setup failed: Cannot update policy for ${POLICY_NAME}"
       exit 1
     fi
 fi
 
-# declare the service as http
-cat <<EOF | consul config write -
-{
-  "kind": "service-defaults",
-  "name": "carbonio-proxy",
-  "protocol": "http"
-}
-EOF
+trap 'echo Script for ${SERVICE_COMMON_NAME} terminated with error' EXIT
+set -e
+# Declare the service protocol
+consul config write "${SERVICE_BASE_DIR}/service-protocol-${SERVICE}.json"
 
 if [[ ! -f "/etc/zextras/carbonio-proxy/token" ]]; then
-    # create the token
-    consul acl token create -format json -policy-name "${POLICY_NAME}" -description "Token for carbonio-proxy/$(hostname -A)" |
+    # Create the token
+    consul acl token create -format json -policy-name "${POLICY_NAME}" -description "Token for
+    ${SERVICE_NAME}/$(hostname -A)" |
       jq -r '.SecretID' > /etc/zextras/carbonio-proxy/token;
-    chown carbonio-proxy:carbonio-proxy /etc/zextras/carbonio-proxy/token
+    chown ${SERVICE_USER}:${SERVICE_GROUP} /etc/zextras/carbonio-proxy/token
```

Comment:
Token should be under /etc/carbonio/proxy/service-discover/token followin guidelines

Reply:
ok